tsdb/engine/tsm1/compact.go

```diff
@@ -580,11 +580,24 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) ([]CompactionGroup, int64) {
 	// each generation in descending break once we see a file less than 4.
 	end := 0
 	start := 0
+	lastHighLevelIndex := -1
+
 	for i, g := range generations {
-		if g.level() <= 3 {
+		// Track the last high-level generation
+		if g.level() > 3 {
+			lastHighLevelIndex = i
+		}
+
+		// Skip low-level generations with too many files for level compaction
+		if g.level() <= 3 && len(g.files) >= 4 {
```

Comment:
Isn't `Plan` for full compactions, not level compactions?  I would expect something like this check in `PlanLevel` but not in `Plan`?

Reply:
Perhaps I am misunderstanding something.

Reply:
This check ensures that we will pick up the rogue lower level files if they appear.